### ISSUE TITLE
_sslverify's fallback for service_identity thinks that X509Names are bytes but actually they're unicode

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7094]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#7094|
|Type|defect|
|Created|2014-04-07 18:35:27Z|
|Branch|https://github.com/twisted/twisted/tree/verify-unicode-7094|

This causes the one python 3 builder with pyOpenSSL installed (`fedora-fedora19-x86_64-1`; `fedora-fedora19-x86_64-2` does not) to fail sometimes.

<details><summary>Searchable metadata</summary>

```
trac-id__7094 7094
type__defect defect
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__branches_verify_unicode_7094 branches/verify-unicode-7094
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1396895727000000 1396895727000000
changetime__1396903030000000 1396903030000000
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [42122]) Branching to verify-unicode-7094.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

Build is kicked off.  I anticipate success, although we should make sure it hits the builder (we really shouldn't have randomly differently configured builders in the rotation for the same build).

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|Alex set owner to @glyph|
|-|-|

This patch looks correct to me (assuming tests pass), but it definitely requires a test.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph removed owner|
|-|-|

Here it is with a test that fails if it's broken.

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|Alex set owner to @glyph|
|-|-|

Assuming tests pass, this looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set status to `closed`|
|-|-|

(In [42130]) Merge verify-unicode-7094

Author: glyph

Reviewer: alex

Fixes: [#7094](https://github.com/twisted/twisted/issues/7094)

Fix a minor issue with the #5091 merge; pass tests on python 3 when pyOpenSSL is installed but service_identity isn't.
